clients/src/main/java/org/apache/kafka/common/requests/DescribeQuorumResponse.java

```diff
@@ -93,4 +97,57 @@ public static DescribeQuorumResponseData singletonResponse(TopicPartition topicP
     public static DescribeQuorumResponse parse(ByteBuffer buffer, short version) {
         return new DescribeQuorumResponse(new DescribeQuorumResponseData(new ByteBufferAccessor(buffer), version));
     }
+
+    public String getTopicNameByIndex(Integer index) {
+        return data.topics().get(index).topicName();
+    }
+
+    public Integer getPartitionLeaderId(String topicName, Integer partition) {
+        Integer leaderId = -1;
+        TopicData topic = data.topics().stream()
+            .filter(t -> t.topicName().equals(topicName))
+            .findFirst()
+            .orElse(null);
+        if (topic != null) {
+            leaderId = Integer.valueOf(topic.partitions().get(partition).leaderId());
+        }
+        return leaderId;
+    }
+
+    public List<QuorumInfo.ReplicaState> getVoterInfo(String topicName, Integer partition) {
+        List<QuorumInfo.ReplicaState> voterInfo = new ArrayList<>();
+        TopicData topic = data.topics().stream()
+            .filter(t -> t.topicName().equals(topicName))
+            .findFirst()
+            .orElse(null);
+        if (topic != null) {
+            topic.partitions().get(partition).currentVoters().forEach(
+                v -> {
+                    voterInfo.add(new QuorumInfo.ReplicaState(v.replicaId(),
+                            v.logEndOffset(),
+                            OptionalLong.of(v.lastFetchTimestamp()),
+                            OptionalLong.of(v.lastCaughtUpTimestamp())));
+                });
+        }
+        return voterInfo;
+    }
+
+    public List<QuorumInfo.ReplicaState> getObserverInfo(String topicName, Integer partition) {
```

Comment:
nit: `getVoterInfo` and `getObserverInfo` seem basically the same. Feels like we can factor out the common logic.